### PR TITLE
MAKE-955: update buildlogger cedar protobuf and add options reader

### PR DESCRIPTION
**Pull request by julianedwards**

### CODE REVIEW

File: options_read.go [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+type unmarshaler func([]byte, interface{}) error
+
+func getUnmarshaler(fn string) unmarshaler {
+	switch {
+	case strings.HasSuffix(fn, ".json"):
+		return json.Unmarshal
+	case strings.HasSuffix(fn, ".yaml"), strings.HasSuffix(fn, ".yml"):
+		return yaml.Unmarshal
+	default:
```
Comment (tychoish) [inline options_read.go]: add bson for funzies?

File: basic_sender.go [hunk 2]
```diff
@@ -336,7 +331,7 @@ func (b *buildlogger) Close() error {
 	return catcher.Resolve()
 }
 
-func (b *buildlogger) createNewLog(ts time.Time) error {
+func (b *buildlogger) createNewLog() error {
```
Comment (tychoish) [inline basic_sender.go]: does the service now construct its own time?

Comment (julianedwards) [reply basic_sender.go]: yes

File: options_read.go (hunk 1 above)
Comment (julianedwards) [reply options_read.go]: why not lol

**Pull request merged**